src/mjolnir/elevationbuilder.cc

```diff
@@ -34,6 +34,148 @@ constexpr double POSTING_INTERVAL = 60;
 // Do not compute grade for intervals less than 10 meters.
 constexpr double kMinimumInterval = 10.0f;
 
+typedef std::unordered_map<uint32_t, std::tuple<uint32_t, uint32_t, float, float, float, float>>
+    cache_t;
+
+std::unordered_set<PointLL> get_coord(const std::string& tile_dir, const std::string& tile) {
+  valhalla::mjolnir::GraphTileBuilder tilebuilder(tile_dir, GraphTile::GetTileId(tile_dir + tile),
+                                                  true);
+  tilebuilder.header_builder().set_has_elevation(true);
+  uint32_t count = tilebuilder.header()->directededgecount();
+  std::unordered_set<uint32_t> cache;
+  cache.reserve(2 * count);
+  std::unordered_set<PointLL> result;
+  for (uint32_t i = 0; i < count; ++i) {
+    // Get a writeable reference to the directed edge
+    const DirectedEdge& directededge = tilebuilder.directededge_builder(i);
+    // Get the edge info offset
+    uint32_t edge_info_offset = directededge.edgeinfo_offset();
+    // Check if this edge has been cached (based on edge info offset)
+    if (cache.count(edge_info_offset))
+      continue;
+
+    cache.insert(edge_info_offset);
+    // Get the shape and length
+    auto shape = tilebuilder.edgeinfo(&directededge).shape();
+    auto length = directededge.length();
+    if (!directededge.tunnel() && directededge.use() != Use::kFerry) {
+      // Evenly sample the shape. If it is really short or a bridge just do both ends
+      std::vector<PointLL> resampled;
+      if (length < POSTING_INTERVAL * 3 || directededge.bridge()) {
+        resampled = {shape.front(), shape.back()};
+      } else {
+        resampled = valhalla::midgard::resample_spherical_polyline(shape, POSTING_INTERVAL);
+      }
+      for (auto&& el : resampled)
+        result.insert(std::move(el));
+    }
+  }
+  return result;
+}
+
+void add_elevations_to(GraphReader& graphreader,
```

Comment:
the naming here is a bit odd, maybe `add_elevation_to_tile`?

Reply:
yeah, "add_elevation_to_tile" sounds much better